src/query/users/src/jwt/authenticator.rs

```diff
@@ -68,21 +77,31 @@ impl JwtAuthenticator {
         if jwt_key_file.is_empty() {
             return Ok(None);
         }
-        let mut verifier =
-            RemoteJwksVerifier::new(jwt_key_file, None, Duration::from_secs(15 * 60));
-        verifier.set_require_kid(false);
-        Ok(Some(JwtAuthenticator { verifier }))
+        let key_store = jwk::JwkKeyStore::new(jwt_key_file).await?;
+        Ok(Some(JwtAuthenticator { key_store }))
     }
 
-    pub async fn parse_jwt(&self, token: &str) -> Result<HeaderAndClaims<CustomClaims>> {
-        match self.verifier.verify::<CustomClaims>(token).await {
-            Ok(c) => match c.claims().sub {
-                None => Err(ErrorCode::AuthenticateFailure(
-                    "missing field `subject` in jwt",
-                )),
-                Some(_) => Ok(c),
+    pub fn parse_jwt_claims(&self, token: &str) -> Result<JWTClaims<CustomClaims>> {
+        let pub_key = self.key_store.get_key(None)?;
+        match &pub_key {
```

Comment:
what if got a pub key type other than RSA256 and ES256 ? q.q

Reply:
There are only two pub key types for JWK now: https://www.rfc-editor.org/rfc/rfc7518#page-28

Reply:
We will support other `crv` in future PRs.